manager/allocator/network.go

```diff
@@ -992,3 +1090,40 @@ func updateTaskStatus(t *api.Task, newStatus api.TaskState, message string) {
 	t.Status.Message = message
 	t.Status.Timestamp = ptypes.MustTimestampProto(time.Now())
 }
+
+func GetIngressNetwork(s *store.MemoryStore) (*api.Network, error) {
+	var (
+		networks []*api.Network
+		err      error
+	)
+	s.View(func(tx store.ReadTx) {
+		networks, err = store.FindNetworks(tx, store.All)
+	})
+	if err != nil {
+		return nil, err
+	}
+	for _, n := range networks {
+		if isIngressNetwork(n) {
+			return n, nil
+		}
+	}
+	return nil, grpc.Errorf(codes.NotFound, "no ingress network found")
```

Comment:
I think it's better not to return a gRPC error here, because the allocator doesn't use gRPC at all. Maybe define an exported error and return that.

```go
var ErrNoIngressNetwork = errors.New("no ingress network found")
```

Then callers can easily check if this error was returned, and translate to a gRPC error if appropriate.